pkg/controller/disruption/disruption.go

```diff
@@ -326,12 +326,29 @@ func (dc *DisruptionController) implementsScale(gv schema.GroupVersion, kind str
 	if err != nil {
 		return false, err
 	}
+
+	var subresourceName string
 	for _, resource := range resourceList.APIResources {
-		if resource.Kind != kind {
+		if resource.Kind == kind {
+			subresourceName = fmt.Sprintf("%s/scale", resource.Name)
```

Comment:
We'd better check if the `resource.Name` contains "/" to avoid generating `subresourceName` like `{Kind}/status/scale` (even if it seems apiserver will put the sub resources behind their main resource in the `APIResources` list).

Reply:
good catch, will add an extra guard and reorder the test to account for that